### PR TITLE
[Brownfield] Exceptions

### PR DESCRIPTION
**Pull request by annakocheshkova**
Replaced some usages of `AppCenterLog.error()` with real exceptions.

### CODE REVIEW

File: android/app/src/androidTest/java/com/microsoft/codepush/common/utils/FileAndroidTests.java [hunk 1]
```diff
@@ -234,4 +236,41 @@ public void tearDown() throws Exception {
...
+        doReturn(true).when(testFile).isDirectory();
+        mFileUtils.deleteFileOrFolderSilently(testFile);
+    }
+
+    /**
+     * {@link FileUtils#deleteFileOrFolderSilently(File)} should throw a {@link CodePushFileException}
+     * if <code>delete</code> returned <code>null</code>.
+     */
+    @Test(expected = CodePushFileException.class)
+    public void tests2() throws Exception {
```
Comment (ruslan-bikkinin) [inline android/app/src/androidTest/java/com/microsoft/codepush/common/utils/FileAndroidTests.java]: Please use more detailed name.

File: android/app/src/androidTest/java/com/microsoft/codepush/common/utils/FileAndroidTests.java [hunk 2]
```diff
@@ -234,4 +236,41 @@ public void tearDown() throws Exception {
         File testFolder = getTestingDirectory();
         testFolder.delete();
     }
+
+    /**
+     * {@link FileUtils#deleteFileOrFolderSilently(File)} should throw a {@link CodePushFileException}
+     * if <code>listFiles</code> returned <code>null</code>.
+     */
+    @Test(expected = CodePushFileException.class)
+    public void tests() throws Exception {
```
Comment (ruslan-bikkinin) [inline android/app/src/androidTest/java/com/microsoft/codepush/common/utils/FileAndroidTests.java]: Please use more detailed name.

File: android/app/src/androidTest/java/com/microsoft/codepush/common/utils/FileAndroidTests.java [hunk 3]
```diff
@@ -234,4 +236,41 @@ public void tearDown() throws Exception {
...
+        doReturn(false).when(testFile).delete();
+        mFileUtils.deleteFileOrFolderSilently(testFile);
+    }
+
+    /**
+     * {@link FileUtils#deleteFileOrFolderSilently(File)} should throw a {@link CodePushFileException}
+     * if <code>delete</code> on child file returned <code>null</code>.
+     */
+    @Test(expected = CodePushFileException.class)
+    public void tests3() throws Exception {
```
Comment (ruslan-bikkinin) [inline android/app/src/androidTest/java/com/microsoft/codepush/common/utils/FileAndroidTests.java]: Please use more detailed name.

File: android/app/src/main/java/com/microsoft/codepush/common/exceptions/CodePushFileException.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package com.microsoft.codepush.common.exceptions;
+
+/**
+ * Exception class for throwing exceptions occurred when working with files.
+ */
+public class CodePushFileException extends Exception {
```
Comment (ruslan-bikkinin) [inline android/app/src/main/java/com/microsoft/codepush/common/exceptions/CodePushFileException.java]: What is the reason for adding FileException if we already have IOException?

Comment (annakocheshkova) [reply android/app/src/main/java/com/microsoft/codepush/common/exceptions/CodePushFileException.java]: @ruslan-bikkinin Ibelieve its no good to throw an `IOException`, we should throw our own exceptions only.

Comment (annakocheshkova) [reply android/app/src/main/java/com/microsoft/codepush/common/exceptions/CodePushFileException.java]: I think I would change that in other places, too

File: android/app/src/main/java/com/microsoft/codepush/common/managers/CodePushUpdateManager.java [hunk 5]
```diff
@@ -395,10 +395,10 @@ public void unzipPackage(File downloadFile) throws CodePushUnzipException {
         String unzippedFolderPath = getUnzippedFolderPath();
         try {
             mFileUtils.unzipFile(downloadFile, new File(unzippedFolderPath));
+            mFileUtils.deleteFileOrFolderSilently(downloadFile);
```
Comment (ruslan-bikkinin) [inline android/app/src/main/java/com/microsoft/codepush/common/managers/CodePushUpdateManager.java]: What is the reason for moving it here? This method isn't throw no more exception anyway.

Comment (annakocheshkova) [reply android/app/src/main/java/com/microsoft/codepush/common/managers/CodePushUpdateManager.java]: @ruslan-bikkinin now it throws an `IOException`

File: android/app/src/main/java/com/microsoft/codepush/common/utils/FileUtils.java [hunk 6]
```diff
@@ -137,27 +135,26 @@ public void deleteDirectoryAtPath(String directoryPath) throws IOException {
      * @param file path to file/folder to be deleted.
      */
     @SuppressWarnings("WeakerAccess")
-    public void deleteFileOrFolderSilently(File file) {
+    public void deleteFileOrFolderSilently(File file) throws IOException {
```
Comment (ruslan-bikkinin) [inline android/app/src/main/java/com/microsoft/codepush/common/utils/FileUtils.java]: This method should't throw any errors as said in his description , so please remove this change and place TODO comment here in order to don't forget to return here later to add `trackException` instead.

**Pull request merged**